types/src/request.rs

```diff
@@ -255,27 +278,34 @@ mod test {
 		let id = Id::Number(1); // It's enough to check one variant, since the type itself also has tests.
 		let params = Some(RawValue::from_string("[42,23]".into()).unwrap());
 
-		let test_vector: &[(&'static str, Option<_>, Option<_>, &'static str)] = &[
+		let test_vector: &[(&'static str, Option<_>, Option<_>, &'static str, Option<TwoPointZero>)] = &[
 			// With all fields set.
 			(
 				r#"{"jsonrpc":"2.0","id":1,"method":"subtract","params":[42,23]}"#,
 				Some(id.clone()),
 				params.clone(),
 				method,
+				Some(TwoPointZero),
 			),
 			// Escaped method name.
-			(r#"{"jsonrpc":"2.0","id":1,"method":"\"m"}"#, Some(id.clone()), None, "\"m"),
+			(r#"{"jsonrpc":"2.0","id":1,"method":"\"m"}"#, Some(id.clone()), None, "\"m", Some(TwoPointZero)),
 			// Without ID field.
-			(r#"{"jsonrpc":"2.0","id":null,"method":"subtract","params":[42,23]}"#, None, params, method),
+			(
+				r#"{"jsonrpc":"2.0","id":null,"method":"subtract","params":[42,23]}"#,
+				None,
+				params,
+				method,
+				Some(TwoPointZero),
+			),
 			// Without params field
-			(r#"{"jsonrpc":"2.0","id":1,"method":"subtract"}"#, Some(id), None, method),
+			(r#"{"jsonrpc":"2.0","id":1,"method":"subtract"}"#, Some(id), None, method, Some(TwoPointZero)),
 			// Without params and ID.
-			(r#"{"jsonrpc":"2.0","id":null,"method":"subtract"}"#, None, None, method),
+			(r#"{"jsonrpc":"2.0","id":null,"method":"subtract"}"#, None, None, method, Some(TwoPointZero)),
```

Comment:
Was the idea of giving the version too so that you'd add a test which didn't have the `"jsonrpc"` field?